### PR TITLE
Fix most tests and linting errors

### PR DESCRIPTION
**Pull request by WesleydeSouza**
Tests were failing due to linting errors, this fixes most of them (there is still some inconsistent behavior on touch tests).

PhantomJS is still not running properly, not sure why.

### CODE REVIEW

File: lib/events.js [hunk 1]
```diff
@@ -180,7 +181,7 @@ Rye.define('Events', function(){
             EventEmitter.prototype.removeListener.call(this, event, handler)
             if (!this.events[event] && proxy) {
                 this.element.removeEventListener(getType(event), proxy, false)
-                delete this.proxied[event]
+                this.proxied[event] = undefined
```
Comment (ricardobeat) [inline lib/events.js]: Reason for change?

File: lib/query.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ Rye.define('Query', function(){
           , '_': 'querySelectorAll'
         }
       , matchesSelector
+      , dummyDiv = document.createElement('div')
```
Comment (ricardobeat) [inline lib/query.js]: Oops. Was broken in #32, need to fix CI.

File: lib/request.js [hunk 3]
```diff
@@ -194,7 +194,8 @@ Rye.define('Request', function(){
     this.serialize = function () {
         var form = this.get(0)
           , fields = {}
-        new Rye(form && form.elements).forEach(function(field){
+          , ryeForm = new Rye(form && form.elements)
+        ryeForm.forEach(function(field){
```
Comment (ricardobeat) [inline lib/request.js]: superfluous variable?

File: lib/util.js [hunk 4]
```diff
@@ -10,19 +10,20 @@ Rye.define('Util', function(){
         }
 
     function each (obj, fn, context) {
+        var i = null
```
Comment (ricardobeat) [inline lib/util.js]: We don't really follow crock style vars-at-the-top. It mandates a linter to avoid mistakes.

File: lib/events.js (hunk 1 above)
Comment (WesleydeSouza) [inline lib/events.js]: The current linter does not recognize the `delete` keyword.

File: lib/request.js (hunk 3 above)
Comment (WesleydeSouza) [inline lib/request.js]: Another linter complaint.

File: lib/util.js (hunk 4 above)
Comment (WesleydeSouza) [inline lib/util.js]: This variable was declared twice in this function, should it just not have the `var` keyword on the second use?

I believed that just defining it prior to both would look better.

**Comment by ricardobeat** [comment]
Merged manually into master.

**Pull request closed without merging**